lib/chef/provider/package.rb

```diff
@@ -304,6 +310,14 @@ def unlock_package(name, version)
         raise( Chef::Exceptions::UnsupportedAction, "#{self} does not support :unlock" )
       end
 
+      def validate_package(name)
+        raise( Chef::Exceptions::Package, "Package names containing whitespace characters such as spaces, tabs, or newlines are not allowed." ) if valid_package_names?(name)
+      end
+
+      def valid_package_names?(package_name)
+        Array(package_name).any? { |name| name.index(/\s|\\t|\\n/) }
+      end
```

Comment:
So there are concerns that we're injecting this into every package provider now.  Particularly for package_names with source paths to files then there could be files with spaces that this would hit.

It is probably most appropriate to leave the hooks here in the superclass, but leave an empty implementation in the superclass, then move the implementation of the validation code into the yum_package provider.